corgidrp/data.py

```diff
@@ -162,12 +162,13 @@ def __init__(self, data_or_filepath, pri_hdr=None, ext_hdr=None, err = None, dq
 
                 # we assume that if the err and dq array is given as parameter they supersede eventual err and dq extensions
                 if err is not None:
-                    if np.shape(self.data) != np.shape(err):
-                        raise ValueError("The shape of err is {0} while we are expecting shape {1}".format(err.shape, self.data.shape))
+                    if np.shape(self.data) != np.shape(err)[-2:]:
```

Comment:
Rather than passing in err and dq arrays, should we make folks pass in HDUs? That way they automatically come with headers?

Reply:
see my comment above. Well, I guess in the intermediate pipeline steps we return the changed image, only at the end of a level or if wished we save the fits file. We would need a similar function for dq, e.g. if we want to merge a bad pixel mask with the existing dq.

Reply:
It seems like to me that we don't make an image from scratch except at the very beginning. Do we have a use case where we create the full Image class with multiple error maps and an image header? The only one I could imagine is during testing, we want to create data at an intermediate data reduction level. If it's just for testing, the UI/interface for it doesn't need to be as clean as if it's going to be a public-facing use case.